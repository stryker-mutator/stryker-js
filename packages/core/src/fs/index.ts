export * from './file-system.js';
export * from './project.js';
export * from './project-file.js';
export * from './project-reader.js';
