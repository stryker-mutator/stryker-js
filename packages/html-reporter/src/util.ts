import * as path from 'path';
import { promisify } from 'util';

import { fsAsPromised } from '@stryker-mutator/util';
import * as mkdirp from 'mkdirp';
import * as rimraf from 'rimraf';

export function copyFile(fromFilename: string, toFilename: string): Promise<void> {
  return new Promise<void>((resolve, reject) => {
    const readStream = fsAsPromised.createReadStream(fromFilename);
    const writeStream = fsAsPromised.createWriteStream(toFilename);
    readStream.on('error', reject);
    writeStream.on('error', reject);
    readStream.pipe(writeStream);
    readStream.on('end', resolve);
  });
}

export const deleteDir = promisify(rimraf);

export const mkdir = promisify(mkdirp);

export async function writeFile(fileName: string, content: string) {
  await mkdir(path.dirname(fileName));
  await fsAsPromised.writeFile(fileName, content, 'utf8');
}
