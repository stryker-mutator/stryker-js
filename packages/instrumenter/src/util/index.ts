export * from './syntax-helpers.js';
export * from './binary-operator.js';
