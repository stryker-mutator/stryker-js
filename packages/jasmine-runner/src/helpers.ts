import { TestResult, TestStatus } from '@stryker-mutator/api/test-runner';
import jasmine from 'jasmine';

export const helpers = {
  toStrykerTestResult(specResult: jasmine.SpecResult, timeSpentMs: number): TestResult {
    const baseResult = {
      id: specResult.id.toString(),
      name: specResult.fullName,
      timeSpentMs,
    };
    if (specResult.status === 'disabled' || specResult.status === 'pending' || specResult.status === 'excluded') {
      return {
        ...baseResult,
        status: TestStatus.Skipped,
      };
    } else if (!specResult.failedExpectations || specResult.failedExpectations.length === 0) {
      return {
        ...baseResult,
        status: TestStatus.Success,
      };
    } else {
      return {
        ...baseResult,
        status: TestStatus.Failed,
        failureMessage: specResult.failedExpectations.map((failedExpectation) => failedExpectation.message).join(','),
      };
    }
  },
  createJasmine(options: jasmine.JasmineOptions): jasmine {
    return new jasmine(options);
  },
};
