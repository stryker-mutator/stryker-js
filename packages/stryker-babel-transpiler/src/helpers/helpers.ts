import * as path from 'path';

export function toJSFileName(fileName: string) {
  const fileNameInfo = path.parse(fileName);
  const newFileName = path.join(fileNameInfo.dir, fileNameInfo.name + '.js');
  return newFileName;
}
