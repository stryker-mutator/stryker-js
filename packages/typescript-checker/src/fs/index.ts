export * from './hybrid-file-system';
