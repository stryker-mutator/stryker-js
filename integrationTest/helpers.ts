import * as path from 'path';
import * as fs from 'mz/fs';
import { expect } from 'chai';
import { ScoreResult } from 'stryker-api/report';

export async function readScoreResult(eventResultDirectory = path.resolve('reports', 'mutation', 'events')) {
    const allReportFiles = await fs.readdir(eventResultDirectory);
    const scoreResultReportFile = allReportFiles.find(file => !!file.match(/.*onScoreCalculated.*/));
    expect(scoreResultReportFile).ok;
    const scoreResultContent = await fs.readFile(path.resolve(eventResultDirectory, scoreResultReportFile || ''), 'utf8');
    return JSON.parse(scoreResultContent) as ScoreResult;
}