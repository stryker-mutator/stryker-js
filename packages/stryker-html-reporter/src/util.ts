import { fsAsPromised } from '@stryker-mutator/util';
import * as path from 'path';
import * as mkdirp from 'mkdirp';
import * as rimraf from 'rimraf';

async function copyFolderOrFile(fromPath: string, toPath: string): Promise<void> {
  const stats = await fsAsPromised.stat(fromPath);
  if (stats.isDirectory()) {
    return copyFolder(fromPath, toPath);
  }
  else {
    return copyFile(fromPath, toPath);
  }
}

export function copyFolder(fromPath: string, to: string): Promise<void> {
  return mkdir(to)
    .then(() => fsAsPromised.readdir(fromPath))
    .then(files => Promise.all(files.map(file => copyFolderOrFile(path.join(fromPath, file), path.join(to, file)))))
    .then(_ => void 0);
}

function copyFile(fromFilename: string, toFilename: string): Promise<void> {
  return new Promise<void>((resolve, reject) => {
    const readStream = fsAsPromised.createReadStream(fromFilename);
    const writeStream = fsAsPromised.createWriteStream(toFilename);
    readStream.on('error', reject);
    writeStream.on('error', reject);
    readStream.pipe(writeStream);
    readStream.on('end', () => resolve());
  });
}

export function deleteDir(dirToDelete: string): Promise<void> {
  return new Promise((res, rej) => {
    rimraf(dirToDelete, err => {
      if (err) {
        rej(err);
      } else {
        res();
      }
    });
  });
}

export function mkdir(folderName: string): Promise<void> {
  return new Promise<void>((res, rej) => {
    mkdirp(folderName, err => {
      if (err) {
        rej(err);
      } else {
        res();
      }
    });
  });
}

export function writeFile(fileName: string, content: string) {
  return mkdir(path.dirname(fileName))
    .then(_ => fsAsPromised.writeFile(fileName, content, 'utf8'));
}

export function countPathSep(fileName: string) {
  let count = 0;
  for (const ch of fileName) {
    if (ch === path.sep) {
      count++;
    }
  }
  return count;
}
