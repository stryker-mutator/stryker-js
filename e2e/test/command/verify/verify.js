import { promises as fsPromises } from 'fs';

import { expect } from 'chai';

import { expectMetricsJsonToMatchSnapshot } from '../../../helpers.js';

describe('After running stryker with the command test runner', () => {
  it('should report 64% mutation score', async () => {
    await expectMetricsJsonToMatchSnapshot();
  });
  it('should write to a log file', async () => {
    const strykerLog = await fsPromises.readFile('./stryker.log', 'utf8');
    expect(strykerLog).contains('INFO DryRunExecutor Initial test run succeeded. Ran 1 test');
    expect(strykerLog).matches(/MutationTestExecutor Done in \d+/);
    expect(strykerLog).not.contains('ERROR');
    expect(strykerLog).not.contains('WARN');
  });
});
