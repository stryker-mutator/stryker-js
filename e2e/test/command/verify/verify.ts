import { fsAsPromised } from '@stryker-mutator/util';
import { expect } from 'chai';
import { expectMetricsResult, produceMetrics } from '../../../helpers';

describe('After running stryker with the command test runner', () => {
  it('should report 64% mutation score', async () => {
    await expectMetricsResult({
      metrics: produceMetrics({
        killed: 16,
        mutationScore: 64,
        mutationScoreBasedOnCoveredCode: 64,
        survived: 9,
        totalCovered: 25,
        totalDetected: 16,
        totalMutants: 25,
        totalUndetected: 9,
        totalValid: 25
      })
    });
  });

  it('should write to a log file', async () => {
    const strykerLog = await fsAsPromised.readFile('./stryker.log', 'utf8');
    expect(strykerLog).contains('INFO InitialTestExecutor Initial test run succeeded. Ran 1 test');
    expect(strykerLog).matches(/Stryker Done in \d+/);
    expect(strykerLog).not.contains('ERROR');
  });
});
