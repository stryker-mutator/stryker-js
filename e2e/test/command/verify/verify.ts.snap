// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`After running stryker with the command test runner should report 64% mutation score 1`] = `
Object {
  "compileErrors": 0,
  "ignored": 0,
  "killed": 16,
  "mutationScore": 64,
  "mutationScoreBasedOnCoveredCode": 64,
  "noCoverage": 0,
  "runtimeErrors": 0,
  "survived": 9,
  "timeout": 0,
  "totalCovered": 25,
  "totalDetected": 16,
  "totalInvalid": 0,
  "totalMutants": 25,
  "totalUndetected": 9,
  "totalValid": 25,
}
`;
